q2_chemistree/_collate_fingerprint.py

```diff
@@ -48,15 +56,25 @@ def collate_fingerprint(csi_result: CSIDirFmt) -> biom.Table:
                 with open(os.path.join(fidpath, 'fingerprints', fname)) as f:
                     fp = f.read().strip().split('\n')
                 molfp[fid] = fp
-
     fingerids = pd.DataFrame.from_dict(molfp, orient='index')
     if fingerids.shape == (0, 0):
         raise ValueError('Fingerprint file is empty!')
+    substructrs = pd.read_table(os.path.join(csi_result, 'fingerprints.csv'),
+                                index_col='absoluteIndex')
     fingerids.index.name = '#featureID'
+    fingerids.columns = substructrs.index
```

Comment:
Is there a guarantee that the the `fingerids` table is aligned with the `substructrs` table? Otherwise this might just scramble the names of the positions. I would suggest adding an additional step where this alignment is forced through the `relativeindex` column.